jupyter_ascending/requests/execute.py

```diff
@@ -58,4 +59,6 @@ def send(file_name: str, line_number: int, *args, **kwargs):
 
     arguments = parser.parse_args()
 
+    # Sync code first
+    sync_send(arguments.filename)
     send(arguments.filename, arguments.linenumber)
```

Comment:
this seems racy. Consider this execution order:

1. User calls this script, and the `sync_send` line runs to completion
2. User separately changes the code and then calls a command to sync code. That code runs to completion
3. The `send` line here executes

That means that the NEW code will be executed, ie, the execute request is not atomic.

In practice it's probably not a huge deal, but it seems like it would both be safer and simpler if we had only a single request type, and that request executed atomically (either to update the code and then execute the cell, in the case or this script, or to just update the code, in the case of the other script)

Does that seem better to you?

Reply:
Even execute_all could be combined into the same request. Then it would all be simpler and atomic--a request to the server will do these:

1. (always) Update the code
2. (optionally) execute some cells (either one or all)

Reply:
I agree this would be better - but would require a (small) refactor to change how the client, the JsonRequest data types, and the notebook extension handle requests (would need to pass in the whole file for all of them). In practice idk that anyone would notice the difference. Happy to change it though if you think it's worth.

Reply:
Let's just make an issue for it and come back to it if we want to in the future (if people actually end up using this project, it likely is worth it, but we can wait)

Reply:
Great, opened #20